Support for abstract base models improved. (#516)

diegocastrum opened the pull request:
Model classes moved into a `models.py` directory, which exports by default only the default `django-celery-beat` model classes now from `models.generic.py`.

`managers.py` moved into `models.py` directory, because it's something that is needed only by models.

diegocastrum committed:
Support for abstract base models improved. (#516)

Model classes moved into a `models.py` directory, which exports by
default only the default `django-celery-beat` model classes now from
`models.generic.py`.

`managers.py` moved into `models.py` directory, because it's something
that is needed only by models.

SimeonAleksov commented:
@auvipy any plans to merge this soon?

auvipy commented:
no, there are other priorities for a long overdue release

diegocastrum committed:
Exporting 'crontab_schedule_celery_timezone' from 'models.abstract'

`crontab_schedule_celery_timezone` wasn't exported properly and
raised an `AttributeError` when applying migrations.

diegocastrum committed:
Fixed a wrong export introduced in prev commit

diegocastrum committed:
Bypass de default models used by `django_celery_beat.scheduler` (#516)

when
`CELERY_BEAT_(?:PERIODICTASKS?|(?:CRONTAB|INTERVAL|SOLAR|CLOCKED)SCHEDULE)_MODEL`
constants are defined in django `settings`.

Providing the `app_label.model_name` of your own models as value for the
constants
`CELERY_BEAT_(?:PERIODICTASKS?|(?:CRONTAB|INTERVAL|SOLAR|CLOCKED)SCHEDULE)_MODEL`
will let `django_celery_beat.scheduler` use the custom models instead of
the default ones (aka generic models, in this context/pull-request):

```python
CELERY_BEAT_PERIODICTASK_MODEL = "app_label.model_name"
CELERY_BEAT_PERIODICTASKS_MODEL = "app_label.model_name"
CELERY_BEAT_CRONTABSCHEDULE_MODEL = "app_label.model_name"
CELERY_BEAT_INTERVALSCHEDULE_MODEL = "app_label.model_name"
CELERY_BEAT_SOLARSCHEDULE_MODEL = "app_label.model_name"
CELERY_BEAT_CLOCKEDSCHEDULE_MODEL = "app_label.model_name"
```

Doing this we add support to automatically bypass the default
`django_celery_beat` models without forcing developers to overwrite the
whole `django_celery_beat.scheduler` in projects where the default
models doesn't fit the requirements

I updated the `README.rst` with a small explanation about how this work

Additonal information:
* related issue: #516
* pull-request: #534

kevindice commented:
I do think this would be useful - for example, I'd like to add a description field to have a human-readable description for the interval/cron/solar schedule.

cclauss commented:
Git conflicts — please rebase.

diegocastrum commented:
I will try to do it asap

diegocastrum committed:
Add ClockScheduler model to manage scheduled tasks

diegocastrum committed:
Remove ClockScheduler model from migrations and generic models

diegocastrum committed:
Refactor models and documentation for custom periodic tasks support

Fixes #516

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

diegocastrum committed:
Refactor import statements for improved readability and consistency

Fixes #911

diegocastrum commented:
@cclauss I’ve rebased this PR on top of `celery/django-celery-beat:main` and did a cleanup pass to make the branch easier to review.

Summary of what changed:

- Rebased the old abstract-models work onto the latest `main`.
- Removed an accidental/stale `ClockScheduler` model and its generated migration.
- Kept the default `django_celery_beat` model behavior intact.
- Centralized custom model resolution through helper functions.
- Updated signal registration and scheduler model resolution so custom models are used consistently.
- Preserved compatibility imports from `django_celery_beat.models`.
- Added focused tests for custom model resolution and signal wiring.
- Updated README/docs with custom model guidance.
- Added API reference docs for the new split model modules.

@auvipy I’d appreciate another review when someone has time. Thanks!

diegocastrum committed:
Improve error messages in _configured_model for better clarity

diegocastrum committed:
Fix lazy scheduler model resolution
